### PR TITLE
[Breadcrumb] Interaction tests

### LINKED ISSUE #3979: [Breadcrumb] Missing interaction tests
**Describe the bug**
No interaction tests were added to the first merged version of Breadcrumb. Those tests are important to verify the functionality of the ellipsis item and the raising of the ItemClicked event

### PR DESCRIPTION
**Pull request by almedina-ms**
## Description

This PR includes the first interaction tests for the Breadcrumb control, the tests currently verify the following scenarios:
| Test | Result |
| --- | --- |
| Default control rendering | No interaction is performed on the control, verifies control initial state |
| Adding elements to the ItemsSource | Verifies the number of breadcrumb items increase |
| Invoking one of the breadcrumb items | Verifies the list of breadcrumb items change once an item has been invoked |
| Adding elements to the ItemsSource and limiting the rendering space | Verifies that the ellipsis items repeater gets populated |
| Invoking one of the ellipsis items | Verifies the breadcrumb ItemsSource gets updated |

> NOTE: this is a work in progress as keyboard interaction has not been tested yet

Besides that, some changes required for the interaction tests have been added into the breadcrumb control, mainly providing accessible names to parts of the control

## Motivation and Context
Fixes #3979

## How Has This Been Tested?
This change includes tests so /azp run should be run when merging the change

### CODE REVIEW

File: dev/Breadcrumb/Breadcrumb.xaml [hunk 1]
```diff
@@ -17,7 +17,7 @@
             <Setter.Value>
                 <ControlTemplate TargetType="local:BreadcrumbItem">
 
-                    <Button x:Name="PART_BreadcrumbItemButton">
+                    <Button x:Name="PART_BreadcrumbItemButton" AutomationProperties.Name="BreadcrumbItemButton">
```
Comment (chingucoding) [inline dev/Breadcrumb/Breadcrumb.xaml:20]: I think this should be a localized resource.

File: dev/Breadcrumb/BreadcrumbItem.cpp [hunk 2]
```diff
@@ -127,6 +127,9 @@ void BreadcrumbItem::OnFlyoutElementPreparedEvent(winrt::ItemsRepeater sender, w
     {
         element.PreviewKeyDown({ this, &BreadcrumbItem::OnFlyoutElementKeyDownEvent });
     }
+
+    hstring name = L"EllipsisItem" + winrt::to_hstring(args.Index() + 1);
```
Comment (StephenLPeters) [inline dev/Breadcrumb/BreadcrumbItem.cpp:131]: use `auto const` for local variable definitions when possible

File: dev/Breadcrumb/BreadcrumbItem.cpp [hunk 3]
```diff
@@ -299,6 +302,8 @@ void BreadcrumbItem::InstantiateFlyout()
     {
         // Create ItemsRepeater and set the DataTemplate 
         const auto& ellipsisItemsRepeater = winrt::ItemsRepeater();
+        ellipsisItemsRepeater.Name(L"PART_EllipsisItemsRepeater");
+        winrt::AutomationProperties::SetName(ellipsisItemsRepeater, L"EllipsisItemsRepeater");
```
Comment (StephenLPeters) [inline dev/Breadcrumb/BreadcrumbItem.cpp:306]: This looks very strange, why are we setting an AutomationName from code behind in a keyDownEvent?

File: dev/Breadcrumb/Breadcrumb.xaml (hunk 1 above)
Comment (StephenLPeters) [reply dev/Breadcrumb/Breadcrumb.xaml:20]: Yes, all AutomationProperties.Name values need to be localized, Which I believe means you will need to set this from code behind.  take a look at this for an example: https://github.com/microsoft/microsoft-ui-xaml/blob/a146dd3db9d6b78a46521099944b74934a9be650/dev/NumberBox/NumberBox.cpp#L142

**Review by StephenLPeters — CHANGES_REQUESTED** [review]

File: dev/Breadcrumb/InteractionTests/BreadcrumbTests.cs [hunk 4]
```diff
@@ -1,11 +1,11 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+
```
Comment (beervoley) [inline dev/Breadcrumb/InteractionTests/BreadcrumbTests.cs:5]: nit: extra space

File: dev/Breadcrumb/InteractionTests/BreadcrumbTests.cs [hunk 5]
```diff
@@ -43,9 +44,218 @@ public void TestCleanup()
...
-        public void BasicTest()
+        [TestProperty("TestSuite", "A")]
+        public void NoInteractionTest()
+        {
+            using (var setup = new TestSetupHelper("Breadcrumb Tests"))
+            {
+                Log.Comment("Retrieve breadcrumb control as generic UIElement");
+                UIObject breadcrumb = FindElement.ByName("BreadcrumbControl");
+                Verify.IsNotNull(breadcrumb, "Verifying that we found a UIElement called BreadcrumbControl");
+
```
Comment (beervoley) [inline dev/Breadcrumb/InteractionTests/BreadcrumbTests.cs:55]: Finding BreadCrumbControl is the first thing you do in every test. You can extract these two lines into a method in order not to repeat them every time.

**Pull request closed without merging**